src/serialize_stl.hh

```diff
@@ -63,6 +64,19 @@ template<typename T> struct serialize_as_collection<cb_queue<T>>
 	}
 };
 
+template<typename T, size_t N> struct serialize_as_collection<static_vector<T, N>>
+	: serialize_as_stl_collection<static_vector<T, N>>
+{
+	static constexpr bool loadInPlace = true;
+	static void prepare(static_vector<T, N>& v, int n) {
+		assert(n > 0 && size_t(n) <= N);
+		v.clear();
+	}
+	static T* output(static_vector<T, N>& v) {
+		return v.data();
+	}
```

Comment:
I think this is wrong (saving is fine, but loading is not).
`prepare()` calls `clear()`, making the `static_vector` have size 0. But there's nothing in the code that later set the size to the correct value.

For the other containers `serialize_as_collection::output()` returns a `std::back_inserter()`, that one will call `push_back()`, and thus the size gets increased incrementally.

I see also that you've set `loadInPlace = true`. When possible that's a good idea. But I'm not sure that's compatible with `std::back_inserter()` (or maybe it is, I'm just not sure). (Detail: the effect of `loadInPlace` is implemented in `serialize_core.hh` lines 719-742).

I see two possible solutions:
* The easiest solution is to not use `loadInPlace` (I mean keep the default value `false`). And then you can also keep the default implementation for `prepare()` and `output()`. (Maybe you could still do the `assert()` you have now in `prepare()`).
* If you want to use `loadInPlace=true` you could in `prepare()` replace the current `clear()` with `resize(n)`. Though that requires you also implement `static_vector::resize()`. And that method should must default-initialize the new elements (but these elements will be overwritten again so that's useless work).

Other alternatives could be to extend `static_vector` with a non-standard method to force setting the size, without changing the content of the internal array. But I'd prefer not to do that because that will make it more difficult to in the future switch to `std::inplace_vector`.

Because this code is not performance critical I'd suggest to go for the easiest solution (which is not using `loadInPlace`). That's also how we current serialize a regular `std::vector`.

Reply:
Yeah you’re right there’s a bug.

I’ll just remove the load in place as you say, then the implementation becomes trivial.

Reply:
> * (Maybe you could still do the `assert()` you have now in `prepare()`).

`push_back()` already has an assert in it so not needed.